### PR TITLE
Fixed bug in regression problem

### PR DESCRIPTION
**Pull request by orionw**
- Shuffled data in `predictions` did not match up with the full data that was being passed to the `_score` function.  This fixes it for the neural networks and keeps it the same for the other regression problems.
- Fixes RMSE bug

### CODE REVIEW

File: dna/models.py [hunk 1]
```diff
@@ -379,7 +378,7 @@ def predict_regression(self, data, *, batch_size, verbose):
         data_loader = self._get_data_loader(data, batch_size, False)
         predictions, targets = self._predict_epoch(data_loader, self._model, verbose=verbose)
 
-        return predictions
+        return predictions, targets
```
Comment (orionw) [inline dna/models.py]: This is the change that breaks it.  However, the predictions are shuffled so passing them up and evaluating them on the unchanged full dataset will give us different results.

File: dna/problems.py [hunk 2]
```diff
@@ -56,10 +56,11 @@ class RegressionProblem(ProblemBase):
 
     @staticmethod
     def _score(predictions, data):
```
Comment (orionw) [inline dna/problems.py]: See also this line.  We would have to change this function to account for that.

**Comment by bjschoenfeld** [comment]
We can turn off shuffling in the predict methods

**Comment by bjschoenfeld** [comment]
@orionw Can you finish fixing the bugs you have found so far to finish this PR? As you have time, we can keep looking for other bugs and make new PRs.

**Comment by orionw** [comment]
Yeah I'll wrap this up today - it's a small PR

**Comment by bjschoenfeld** [comment]
@orionw When I run this code, I still see problems. The per_primitive_regression RMSE scores are about 1.3 and 1.6. They should be between 0 and 1. The dna_regression train loss and train score do not match either (validation scores do match).

**Pull request merged**